### PR TITLE
Fix and streamline jet veto maps selection.

### PR DESCRIPTION
**Pull request by riga**
This PR updates the jet veto map selector:

- fixed (flipped) veto map check.
- moved 2023 bpix check to init.
- used `run` and `postfix` aux entries of campaigns.
- merged `veto_map_result` and `veto_map_jet_mask` which were identical, and removed latter from the aux result since it's already stored inside the events chunk.

### CODE REVIEW

File: columnflow/selection/cms/jets.py [hunk 1]
```diff
@@ -53,61 +54,64 @@ def jet_veto_map(
...
-    # get the Jet veto mask (events containing such a jet should be vetoed)
-    veto_map_jet_mask = (veto_map_result > 0)
-
-    if self.config_inst.campaign.x("postfix", "").lower() == "bpix":
-        # in postBPix, we need to run the veto map with type=jetvetomap_bpix and subtract this from
-        # the result of the nominal jet veto map
-        raise NotImplementedError("Jet Veto Map for 2023 postBPix not implemented yet")
+    # apply the veto map to selected jets
+    # (a map value of 0 means the jet is to be vetoed)
+    veto_mask = jet_mask & (self.veto_map(*inputs) == 0)
```
Comment (mafrahm) [inline columnflow/selection/cms/jets.py:90]: No, this is wrong. An event is to be vetoed when a jet with veto map value > 0 survives the jet preselection.
```suggestion
    # (a map value of >0 means the jet is to be vetoed)
    bad_jet = jet_mask & (self.veto_map(*inputs) > 0)
    good_jet = jet_mask | (self.veto_map(*inputs) == 0)
```

Afterwards, we need to define the selection step accordingly
```
# check that no bad jets are present or that all jets are good jets.
jet_veto_map_step = ak.sum(bad_jet, axis=1) == 0
jet_veto_map_step = ak.sum(good_jet, axis=1) == ak.num(good_jet)
```

**Review by mafrahm — CHANGES_REQUESTED** [review]
One very important inline comment regarding the result. Everything else looks good to me

File: columnflow/selection/cms/jets.py (hunk 1 above)
Comment (riga) [reply columnflow/selection/cms/jets.py]: I fixed it, please have a look :+1:

File: columnflow/selection/cms/jets.py [hunk 2]
```diff
@@ -53,61 +54,69 @@ def jet_veto_map(
...
+    # evalute the veto map only for selected jets
+    # (a map value of != 0 means the jet is vetoed)
+    veto_mask = jet_mask
+    flat_veto_mask = flat_np_view(veto_mask)
+    flat_veto_mask[flat_veto_mask] = ak.flatten(self.veto_map(*inputs) != 0)
 
-    # get the Jet veto mask (events containing such a jet should be vetoed)
-    veto_map_jet_mask = (veto_map_result > 0)
+    # store the per-jet veto mask
+    events = set_ak_column(events, "Jet.veto_map_mask", veto_mask)
```
Comment (mafrahm) [inline columnflow/selection/cms/jets.py:96]: to me it looks like this is still just the `jet_mask`. I think you need to use the `layout_ak_array` function to transform the `flat_veto_mask` back into the correct shape.

Comment (riga) [reply columnflow/selection/cms/jets.py:96]: The `flat_veto_mask` is really just a view, so values changed in the view directly translate to changed values in the original, structured `veto_mask`.

**Review by mafrahm — APPROVED** [review]

**Pull request merged**